src/scripts/domain/moveCalculator.ts

```
@@ -225,11 +229,351 @@ export class MoveCalculator {
+      (cell2 !== WallCell && (cell2 === piece || toRemove.includes(cell2)));
+
+    if (hasAttack) {
+      toRemove.push(cell1);
+
+      const partPos = pos.clone();
+      partPos.x -= direction.x * cellsAdvanced;
+      partPos.y -= direction.y * cellsAdvanced;
+
+      path.push(partPos);
+
+      this.addKingAttackMovesRec(
```

Biome (1.9.4)

[error] Using this in a static context can be confusing.

this refers to the class.
Unsafe fix: Use the class name instead.


(lint/complexity/noThisInStatic)